File: backend/app/models/comment.py
"""
评论模型模块

本模块定义了评论相关的数据模型。
"""
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.database import Base


class Comment(Base):
    """
    评论表
    
    存储用户对文章的评论信息。
    
    Attributes:
        commentid (int): 评论 ID，主键
        userid (int): 用户 ID，外键关联 users 表
        articleid (int): 文章 ID，外键关联 article 表
        content (str): 评论内容
        ipaddr (str): 评论者 IP 地址
        replyid (int): 回复的评论 ID (如果是回复)
        agreecount (int): 点赞数
        opposecount (int): 反对数
        hidden (int): 是否隐藏 (0: 否, 1: 是)
        createtime (datetime): 创建时间
        updatetime (datetime): 更新时间
    """
    __tablename__ = "comment"
    
    commentid = Column(Integer, primary_key=True, autoincrement=True)
    userid = Column(Integer, ForeignKey("users.userid"), nullable=False, index=True)
    articleid = Column(Integer, ForeignKey("article.articleid"), nullable=False, index=True)
    content = Column(Text(65536), nullable=False)
    ipaddr = Column(String(30))
    replyid = Column(Integer)
    agreecount = Column(Integer, default=0)
    opposecount = Column(Integer, default=0)
    hidden = Column(Integer, default=0)
    createtime = Column(DateTime, default=datetime.now)
    updatetime = Column(DateTime, default=datetime.now, onupdate=datetime.now)
    
    # 关系
    user = relationship("User", back_populates="comments")
    article = relationship("Article", back_populates="comments")
    
    def __repr__(self):
        return f"<Comment {self.commentid}>"

File: backend_cpp/models/Comment.cc
/**
 * @file Comment.cc
 * @brief Comment Model Implementation
 */

#include "Comment.h"
#include "core/logger.h"

namespace woniunote {
namespace models {

Comment::Comment(const drogon::orm::Row& row)
{
    if (!row["commentid"].isNull()) commentid_ = row["commentid"].as<int64_t>();
    Logger::debug("[Model] Comment loaded", {{"commentid", std::to_string(commentid_)}});
    if (!row["userid"].isNull()) userid_ = row["userid"].as<int64_t>();
    if (!row["articleid"].isNull()) articleid_ = row["articleid"].as<int64_t>();
    if (!row["content"].isNull()) content_ = row["content"].as<std::string>();
    if (!row["ipaddr"].isNull()) ipaddr_ = row["ipaddr"].as<std::string>();
    if (!row["replyid"].isNull()) replyid_ = row["replyid"].as<int64_t>();
    if (!row["agreecount"].isNull()) agreecount_ = row["agreecount"].as<int>();
    if (!row["opposecount"].isNull()) opposecount_ = row["opposecount"].as<int>();
    if (!row["hidden"].isNull()) hidden_ = row["hidden"].as<int>();
}

Json::Value Comment::toJson() const
{
    Json::Value ret;
    ret["commentid"] = static_cast<Json::Int64>(commentid_);
    ret["userid"] = static_cast<Json::Int64>(userid_);
    ret["articleid"] = static_cast<Json::Int64>(articleid_);
    ret["content"] = content_;
    ret["ipaddr"] = ipaddr_;
    ret["replyid"] = static_cast<Json::Int64>(replyid_);
    ret["agreecount"] = agreecount_;
    ret["opposecount"] = opposecount_;
    ret["hidden"] = hidden_;
    return ret;
}

} // namespace models
} // namespace woniunote
